mscore/musescore.cpp

```diff
@@ -587,6 +589,86 @@ void MuseScore::populateNoteInputMenu()
             }
       }
 
+//---------------------------------------------------------
+//   populateFileOperations
+//---------------------------------------------------------
+
+void MuseScore::populateFileOperations()
+      {
+      fileTools->clear();
+
+      if (qApp->layoutDirection() == Qt::LayoutDirection::LeftToRight) {
+            for (auto s : _fileOperationEntries) {
+                  if (!*s)
+                        fileTools->addSeparator();
+                  else
+                        fileTools->addWidget(new AccessibleToolButton(fileTools, getAction(s)));
+                  }
+            }
+      else {
+            _fileOperationEntries.reverse();
+            for (auto s : _fileOperationEntries) {
+                  if (!*s)
+                        fileTools->addSeparator();
+                  else
+                        fileTools->addWidget(new AccessibleToolButton(fileTools, getAction(s)));
+                  }
+            _fileOperationEntries.reverse();
+            }
+
+      // Currently not customizable in ToolbarEditor
+      fileTools->addSeparator();
+      mag = new MagBox;
+      connect(mag, SIGNAL(magChanged(MagIdx)), SLOT(magChanged(MagIdx)));
+      fileTools->addWidget(mag);
+
+      viewModeCombo = new QComboBox(this);
+      #if defined(Q_OS_MAC)
+            viewModeCombo->setFocusPolicy(Qt::StrongFocus);
+      #else
+            viewModeCombo->setFocusPolicy(Qt::TabFocus);
+      #endif
+      viewModeCombo->setFixedHeight(preferences.getInt(PREF_UI_THEME_ICONHEIGHT) + 8);  // hack
+
+      viewModeCombo->setAccessibleName(tr("View Mode"));
+      viewModeCombo->addItem(tr("Page View"), int(LayoutMode::PAGE));
+      viewModeCombo->addItem(tr("Continuous View"), int(LayoutMode::LINE));
+      viewModeCombo->addItem(tr("Single Page"), int(LayoutMode::SYSTEM));
+
+      connect(viewModeCombo, SIGNAL(activated(int)), SLOT(switchLayoutMode(int)));
+      fileTools->addWidget(viewModeCombo);
+      }
+//---------------------------------------------------------
+//   populatePlaybackControls
+//---------------------------------------------------------
+
+void MuseScore::populatePlaybackControls()
+      {
+      transportTools->clear();
+
+      for (const auto s : _playbackControlEntries) {
+            if (!*s)
+                  transportTools->addSeparator();
+            else {
+                  if (QString(s) == "midi-on") {
+                        #ifdef HAS_MIDI
```

Comment:
As I think was mentioned elsewhere, we don't indent ifdef's

Reply:
I asked Jojo Schmitz and he said I should add tabs. Or am I confused about the actual purpose (see an above outdated message for that)

Reply:
if you peruse the source I don't think you'll find a single instance of the ifdef statement itself being indented.  The *content* of the ifdef, sure, it should be inline with the rest of the code.  I think that's what the comment you mention was talking about.  @Jojo-Schmitz?

Reply:
What @Jojo-Schmitz actually wrote was "No indent for pre-processor directives, please.".  Emphasis on the "no".

Reply:
Ah, I was misunderstanding. I thought it meant no indent between the ifdefs. But it meant the ifdefs themselves should have no whitespace before them. I'll fix that